Libraries/LibWeb/IndexedDB/IDBKeyRange.cpp

```diff
@@ -30,4 +37,138 @@ void IDBKeyRange::initialize(JS::Realm& realm)
     WEB_SET_PROTOTYPE_FOR_INTERFACE(IDBKeyRange);
 }
 
+void IDBKeyRange::visit_edges(Visitor& visitor)
+{
+    Base::visit_edges(visitor);
+    visitor.visit(m_lower_bound);
+    visitor.visit(m_upper_bound);
+}
+
+// https://w3c.github.io/IndexedDB/#in
+bool IDBKeyRange::is_in_range(GC::Ref<Key> key) const
+{
+    // A key is in a key range range if both of the following conditions are fulfilled:
+
+    // The range’s lower bound is null, or it is less than key, or it is both equal to key and the range’s lower open flag is false.
+    auto lower_bound_in_range = this->lower_key() == nullptr || Key::less_than(*this->lower_key(), key) || (Key::equals(key, *this->lower_key()) && !this->lower_open());
+
+    // The range’s upper bound is null, or it is greater than key, or it is both equal to key and the range’s upper open flag is false.
+    auto upper_bound_in_range = this->upper_key() == nullptr || Key::less_than(key, *this->upper_key()) || (Key::equals(key, *this->upper_key()) && !this->upper_open());
```

Comment:
I think the `Key::less_than(key, *this->upper_key())` is subtly wrong, as it seems it will return true if upper key is greater than ___or equal to___ key.
```suggestion
    auto upper_bound_in_range = this->upper_key() == nullptr || Key::greater_than(key, *this->upper_key()) || (Key::equals(key, *this->upper_key()) && !this->upper_open());
```

Reply:
Thats true. Seems like I forgot to change this when i moved from operator overloads to methods.. Fixed now 👍